### ISSUE TITLE
Custom Media Directory Error

### ISSUE THREAD

**Issue by TogarUshindi** [AzuraCast/AzuraCast#1944]
**Using Docker installation method**

Yes

**Host Operating System**

Ubuntu Server 18.04.3 LTS

**Describe the bug**

When attempting to create a Custom Media Directory, AzuraCast returns an error:

```
500

Directory "/srv/name/of/full/directory/path" was not created
Station.php : 706 
```

**To Reproduce**

From Dashboard
Click on ADMINISTRATION
Click on Stations
Click on EDIT
Click on ADMINISTRATION
This should take you to /admin/stations/edit/1
In "Advanced Custom Media Directory" form field, type name of custom directory
Click on SAVE CHANGES

**Expected behavior**

Not sure what expected behavior should be.

**Relevant Logs**

Unsure which log to check.

**Device(s):**
 - Device: Desktop
 - OS: Windows 7 (blegh)
 - Browser: Firefox (yay!)
 - Version: 69 (giggity)

**Additional context**

/srv/ is on its own Logical Volume which has much more space than the LV /var is on, so I need my media to be in there.

**Comment by TogarUshindi** [comment]
Maybe this has something to do with the virtual environment and CSRF token issues? Perhaps the server sees it as something on a foreign host and doesn't trust it?

**Comment by Vaalyn** [comment]
Is that path mounted to the AzuraCast stations and web docker containers?

If not then follow this tutorial for mounting volumes: https://www.azuracast.com/help/faq_docker.html#storing-your-station-data-on-the-host-machine

This tutorial shows you how to mount an external path to your containers to store all station data there instead of inside the containers volume.

**Comment by TogarUshindi** [comment]
Ahhh, so, question then: If we must directly create/edit a configuration file, in what circumstances would a user modify the `Advanced Custom Media Directory` field? And what are its limitations? If `/var/azuracast/stations/` must always remain within `/var/` to function correctly, why does it let us modify the path? Or, if not, what are the exceptions?

Perhaps we may want to include a couple new form fields on that page for defining a path to the host so directly editing an override file. I don't know if I should open a feature request since I don't entirely understand why things are done the way they're done. :)

When I enter `/path/on/host/computer:/var/azuracast/stations` in `docker-compose.override.yml`, what happens? Does `/var/azuracast/stations` become a shortcut to `/path/on/host/computer`?

**Comment by SlvrEagle23** [comment]
@TogarUshindi The container that Docker runs is an isolated "sandbox" with its own directory structure and paths that are completely independent of the host server you're hosted on.

The way Docker lets you connect that inside world to the outside world is via those bind mounts, where you link a volume somewhere on the host server to a corresponding internal path inside the container.

The reason we allow you to specify any path in the "Advanced Custom Media Directory" field is because you can mount an external directory anywhere you want in that container, regardless of what its actual path is out on the host computer.

However, we can't actually do the Docker configuration work for you, because AzuraCast's code exists only _inside_ that container, and it isn't aware of the container itself and can't reconfigure it from within itself. This is by design and is usually a security feature, though it involves some measure of inconvenience in advanced use cases like this.

Basically, your "Advanced Custom Media Directory" should match up with the second part of the volume mount; in the example, you're remounting all of `/var/azuracast/stations`, which is the default directory, so you don't really need to customize anything, but if you were to, say, point the volume at `/home/mystation` you could then customize the media directory to point there as well.

This whole Docker paradigm can be understandably confusing, but once things are set up you generally don't have to worry about it too much.

**Comment by TogarUshindi** [comment]
> This whole Docker paradigm can be understandably confusing

It's Rick Sanchez's car battery?
